### PR TITLE
WT-10825 Allow shared operations to use different tables in Workgen

### PR DESCRIPTION
**Pull request by tammybailey**
This PR adds functionality to workgen so that different tables can be assigned to a single operation that is shared among threads.

### CODE REVIEW

**Comment by tammybailey** [comment]
Moving to draft mode while I refactor!

**Review by ddanderson — APPROVED** [review]
Looks good.  A couple of Items to consider.

File: bench/workgen/workgen.cpp [hunk 1]
```diff
@@ -1377,6 +1378,11 @@ ThreadRunner::op_create_all(Operation *op, size_t &keysize, size_t &valuesize)
             else
                 usage_flags |= ThreadRunner::USAGE_WRITE;
             _table_usage[op->_table._internal->_tint] = usage_flags;
+        } else {
+            // Set size of vector storing thread-to-table mappings for the operation.
+            if (op->_tables.size() != _wrunner->_trunners.size()) {
+                op->_tables.resize(_wrunner->_trunners.size());
+            }
```
[the remark this answers is not in the document]
Comment (tammybailey) [reply bench/workgen/workgen.cpp]: Very good point! Done in [685b67e](https://github.com/wiredtiger/wiredtiger/pull/8983/commits/685b67ed73be4c78ac7aec06d9f21a603ea0aef9).

File: bench/workgen/workgen.cpp [hunk 2]
```diff
@@ -1446,23 +1452,53 @@ ThreadRunner::op_get_key_recno(Operation *op, uint64_t range, tint_t tint)
...
+/*
+ * Check if the specified operation has an assigned table. For static tables, this information is
+ * saved in the Operation structure. For dynamic tables, the operation maintains a table assignment
+ * for each thread running the operation.
+ */
+bool
+ThreadRunner::op_has_table(Operation *op) const
+{
+    if (op->_random_table) {
+        return (op->_tables[_number] != std::string());
```
[the remark this answers is not in the document]
Comment (tammybailey) [reply bench/workgen/workgen.cpp]: Agreed, updated in [685b67e](https://github.com/wiredtiger/wiredtiger/pull/8983/commits/685b67ed73be4c78ac7aec06d9f21a603ea0aef9).

**Review by sulabhM — APPROVED** [review]

**Pull request merged**